app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/template.rb

```diff
@@ -47,6 +49,87 @@ def raw_delete_image
     _log.error("image=[#{name}], error: #{e}")
   end
 
+  def self.encrypt_with_aes(creds)
+    cipher = OpenSSL::Cipher.new('aes-256-cbc')
+    cipher.encrypt
+
+    key  = Base64.strict_encode64(cipher.random_key)
+    iv   = Base64.strict_encode64(cipher.random_iv)
+    encr = Base64.strict_encode64(cipher.update(creds.to_json) + cipher.final)
+
+    [encr, key, iv]
+  end
+
+  def self.miq_img_by_ids(provider_id, image_id)
+    powervc = ExtManagementSystem.find(provider_id)
+    powervc.get_image_info(image_id)
+  end
+
+  def self.cos_creds(provider_id)
+    cos = ExtManagementSystem.find(provider_id)
+    cos.cos_creds
+  end
+
+  def self.node_creds(provider_id)
+    powervc = ExtManagementSystem.find(provider_id)
+    endp = powervc.node_endpoint
+    auth = powervc.node_auth
+
+    return endp.hostname, auth.userid, auth.password
+  end
+
+  def self.image_ems_ref(bucket_id)
+    MiqTemplate.find(bucket_id).uid_ems
+  end
+
+  def self.bucket_name(bucket_id)
+    CloudObjectStoreContainer.find(bucket_id).name
+  end
```

Comment:
Most of these methods don't make any sense as public class methods on a `CloudManager::Template`.  They should be private helper methods (maybe grouped in their own module).

Reply:
Thanks! I'll take care of that.

Reply:
@agrare I've grouped  the helper functions into own module. However, I'm note sure how to include them as private static methods into my class without doing things like "class << self". Do you have a suggestion?

Reply:
It's just they're called from "self.raw_import_image" so I can't have them as instance methods.

Reply:
The typical way of doing this is to use `included` from `ActiveSupport::Concern`

Something like this gets you a private class method included from a module
```
module TestMixin
  extend ActiveSupport::Concern

  included do
    private_class_method def self.foo
      :tada
    end 
  end 
end
```

Reply:
I did as you suggested here: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/image_import.rb but I'm getting `[["status", "Error"], ["message", "undefined method `node_creds' ` for some reason. Probably something is still missing.

Reply:
Done. I've checked by trying to call the private class methods using Ruby console and it fails as expected:
`NoMethodError (private method `node_creds' called for #<Class:0x000000000884dc00>)`